### PR TITLE
Add support for old charm store while retrieving the best channel to use.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by mhilton** [comment]
:+1:

File: charmstore_test.go [hunk 1]
```diff
@@ -557,6 +559,42 @@ func (s *charmStoreRepoSuite) TestResolveWithChannel(c *gc.C) {
 	}
 }
 
+var sortChannelsTests = []struct {
```
Comment (rogpeppe) [inline charmstore_test.go]: Perhaps add a test that has some unknown channel too?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/283/
Test PASSed.

**Comment by rogpeppe** [comment]
LGTM with one trivial suggestion.

File: charmstore_test.go (hunk 1 above)
Comment (frankban) [inline charmstore_test.go]: Done.

**Comment by frankban** [comment]
Thank you both for the quick reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/284/
Test PASSed.

**Pull request merged**